### ISSUE TITLE
JSON Serialisation

### ISSUE THREAD

**Issue by burakssen** [c3lang/c3c#1395]
Json parsing is already implemented I assume, I think Json serialisation would be a great addition. Like with swift.

Defining a struct like
```swift
struct Person {
  let name: String
  let age: Int
}
```

JSONSerialization can directly create an Item object from 

```json
{
  "name": "Burak",
  "age": 18
}
```
I am not sure how this can be developed but, I can start something if this is inline with the project.

**Comment by lerno** [comment]
It's quite straightforward with C3 actually. I just haven't bothered writing it as I don't think it's super useful.

**Comment by burakssen** [comment]
Yeah, I think the feature looked cool to me probably because swift has a weird object system 😅

**Comment by fithisux** [comment]
@lerno  : Is C3 going to support more than the standard c library? I think betterC is fine with it. Extras can live in another library.

This one can be ported for example or just used

https://github.com/barrust/c-utils

**Comment by lerno** [comment]
@fithisux there is a standard library in C3, so I don't understand the question.

**Comment by fithisux** [comment]
The idea is not to overload it with many things. Also I was under the impression that it uses only stdc like betterC.

I will delve deeper.

**Comment by lerno** [comment]
No, C3 has a cross platform standard library.

**Comment by joshring** [comment]
A standardised way of serialising JSON would be very helpful, I think it would be a big improvement on having many different ad-hoc solutions for it.

**Comment by Caleb-o** [comment]
There is currently a [JSON encoder](https://c3-lang.org/references/docs/stdlib_refcard/#stdencodingjson) in the std lib. In theory you would just need to make the inverse of this. Unless I missed something here?

**Comment by ManuLinares** [comment]
It's already implemented, this issue is stale.
```c3
import std::io, std::encoding::json;

struct Person
{
    String name;
    int age;
    bool active;
}

fn void main()
{
    Person person = { .name = "Manu", .age = 40, .active = true };
    String s = json::marshal(tmem, person);
    io::printn(s);
}
```
```json
{"name":"Manu","age":40,"active":true}
```

**Comment by lerno** [comment]
Closed since it's implemented.
